src/main/kotlin/com/demonwav/mcdev/i18n/inspections/SuperfluousFormatInspection.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.inspections
+
+import com.demonwav.mcdev.i18n.translations.Translation
+import com.demonwav.mcdev.i18n.translations.Translation.Companion.FormattingError
+import com.demonwav.mcdev.util.runWriteAction
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.JavaElementVisitor
+import com.intellij.psi.PsiCall
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiExpression
+import com.intellij.psi.PsiLiteralExpression
+import com.intellij.psi.PsiReferenceExpression
+import com.intellij.util.IncorrectOperationException
+
+class SuperfluousFormatInspection : TranslationInspection() {
+    override fun getStaticDescription() = "Detect superfluous format arguments for translations"
+
+    override fun buildVisitor(holder: ProblemsHolder): PsiElementVisitor = Visitor(holder)
+
+    private class Visitor(private val holder: ProblemsHolder) : JavaElementVisitor() {
+        override fun visitReferenceExpression(expression: PsiReferenceExpression) {
+            val result = Translation.find(expression)
+            if (result != null && result.foldingElement is PsiCall && result.formattingError == FormattingError.SUPERFLUOUS) {
+                registerProblem(expression, result)
+            }
+        }
+
+        override fun visitLiteralExpression(expression: PsiLiteralExpression) {
+            val result = Translation.find(expression)
+            if (result != null && result.foldingElement is PsiCall && result.formattingError == FormattingError.SUPERFLUOUS) {
+                registerProblem(expression, result, RemoveArgumentsQuickFix(result.foldingElement, result.superfluousVarargStart), ChangeTranslationQuickFix("Use a different translation"))
+            }
+        }
+
+        private fun registerProblem(expression: PsiExpression, result: Translation, vararg quickFixes: LocalQuickFix) {
+            holder.registerProblem(expression, "There are missing formatting arguments to satisfy '${result.text}'", ProblemHighlightType.GENERIC_ERROR, *quickFixes)
+        }
+    }
+
+    private class RemoveArgumentsQuickFix(private val call: PsiCall, private val position: Int) : LocalQuickFix {
```

Comment:
IntelliJ reports a potential memory leak here. I'd prefer if we followed their best practices in this regard - or at least have justification for not following them.